processor/redactionprocessor/processor.go

```diff
@@ -251,6 +260,28 @@ func (s *redaction) processAttrs(_ context.Context, attributes pcommon.Map) {
 	s.addMetaAttrs(ignoring, attributes, "", ignoredKeyCount)
 }
 
+//nolint:gosec
+func (s *redaction) maskValue(val string, regex *regexp.Regexp) string {
+	hashFunc := func(match string) string {
+		switch s.hashFunction {
+		case SHA1:
+			return hashString(match, sha1.New())
+		case SHA3:
+			return hashString(match, sha3.New256())
+		case MD5:
+			return hashString(match, md5.New())
```

Comment:
is it expensive to create the hasher on every call of the function? Is the hasher not reusable? Could it be initialized outside the func?

Reply:
tried it out initializiling the hash functions in the `newRedaction()`function, but for some reason, I am getting invalid hashes in the tests. Looking at the implementations of the `*.New()` methods, it shouldn't be that expensive and as far as I saw the usages, this way it is the mostly used

Reply:
Oh yes, hashers are stateful, you'd need to reset them and they are not threadsafe. So please do keep them this way, sorry.